src/common/Analytics/AnalyticsEvents.ts

```diff
@@ -0,0 +1,80 @@
+import { EventEmitter } from 'events';
+
+export type AnalyticsEventTypeHandlers = {
+  track: (eventName: string, event: unknown) => void;
+  identifyUser: (id: string) => void;
+  userPropertyUpdate: (
+    key: string,
+    value: string | number | boolean | undefined,
+  ) => void;
+  resetUser: () => void;
+};
+
+export type AnalyticsEventTypes = keyof AnalyticsEventTypeHandlers;
+export type AnalyticsEventTypeHandler<T extends AnalyticsEventTypes> =
+  AnalyticsEventTypeHandlers[T];
+
+const emitter = new EventEmitter();
+
+export type Tracker<CustomEventMap extends Record<string, unknown>> = {
+  track<Key extends keyof CustomEventMap>(
+    name: Key,
+    event: CustomEventMap[Key],
+  ): void;
+  identifyUser: AnalyticsEventTypeHandlers['identifyUser'];
+  userPropertyUpdate: AnalyticsEventTypeHandlers['userPropertyUpdate'];
+  resetUser: AnalyticsEventTypeHandlers['resetUser'];
+};
+
+/**
+ * Create a type safe way to emit analytics events. Can do the following actions:
+ *
+ * track: Tracks an event with a given name and payload with values relevant to that event.
+ * identifyUser: Allows you to start tracking a logged in users session by their ID.
+ * userPropertyUpdate: Set a value for the current logged in user.
+ * resetUser: Reset the current user tracking session, for when the user logs out.
+ */
+export const createAnalyticsEmitter = <
+  EventMap extends Record<string, unknown>,
+>(): Tracker<EventMap> => {
+  return {
+    track: (name, payload) => emitter.emit('track', name as string, payload),
+    identifyUser: (id) => emitter.emit('identifyUser', id),
+    userPropertyUpdate: (key, value) =>
```

Comment:
nit: since we already have `identifyUser` and `resetUser`, it seems more idiomatic to call this `updateUser`.